zppy/__main__.py

```diff
@@ -97,29 +100,38 @@ def main():
     if args.last_year:
         config["default"]["last_year"] = args.last_year
 
+    existing_bundles: List[Bundle] = []
+
     # climo tasks
-    climo(config, scriptDir)
+    existing_bundles = climo(config, scriptDir, existing_bundles)
 
     # time series tasks
-    ts(config, scriptDir)
+    existing_bundles = ts(config, scriptDir, existing_bundles)
 
     # tc_analysis tasks
-    tc_analysis(config, scriptDir)
+    existing_bundles = tc_analysis(config, scriptDir, existing_bundles)
 
     # e3sm_diags tasks
-    e3sm_diags(config, scriptDir)
+    existing_bundles = e3sm_diags(config, scriptDir, existing_bundles)
 
     # amwg tasks
-    amwg(config, scriptDir)
+    existing_bundles = amwg(config, scriptDir, existing_bundles)
 
     # mpas_analysis tasks
-    mpas_analysis(config, scriptDir)
+    existing_bundles = mpas_analysis(config, scriptDir, existing_bundles)
 
     # global time series tasks
-    global_time_series(config, scriptDir)
+    existing_bundles = global_time_series(config, scriptDir, existing_bundles)
 
     # ilamb_run tasks
-    ilamb_run(config, scriptDir)
+    existing_bundles = ilamb_run(config, scriptDir, existing_bundles)
+
+    for b in existing_bundles:
+        b.display_dependencies()
+        if not b.dry_run:
+            submitScript(
+                b.bundle_file, b.export, dependFiles=b.dependencies_not_in_bundle_file
```

Comment:
Regarding my comment in `docs/source/dev_guide/testing.rst`: when we submit `bundle3`, it will have a dependency listed in `dependencies_not_in_bundle_file` which is run as part of `bundle1`. As the code currently is, however, it has no idea that this dependency is in `bundle1`.